File: examples/arrays.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int main() {
    int arr[10];
    int i;
    int sum;
    int max;

    cout << "Введите 10 чисел:" << endl;
    for (int i = 1; i <= 10; i++) {
        {
            cout << "arr[" << i << "] = ";
            cin >> arr[(i - 1)];
        }
    }
    sum = 0;
    for (int i = 1; i <= 10; i++) {
        sum = (sum + arr[(i - 1)]);
    }
    max = arr[(1 - 1)];
    for (int i = 2; i <= 10; i++) {
        if ((arr[(i - 1)] > max)) {
            max = arr[(i - 1)];
        }
    }
    cout << "Сумма элементов: " << sum << endl;
    cout << "Максимальный элемент: " << max << endl;
    cout << "Среднее значение: " << (sum / 10) << endl;
    return 0;
}

File: examples/arrays.pas
program ArrayOperations;

var
    arr: array[1..10] of integer;
    i, sum, max: integer;

begin
    { Заполнение массива }
    writeln('Введите 10 чисел:');
    for i := 1 to 10 do
    begin
        write('arr[', i, '] = ');
        readln(arr[i]);
    end;
    
    { Вычисление суммы }
    sum := 0;
    for i := 1 to 10 do
        sum := sum + arr[i];
    
    { Поиск максимума }
    max := arr[1];
    for i := 2 to 10 do
        if arr[i] > max then
            max := arr[i];
    
    { Вывод результатов }
    writeln('Сумма элементов: ', sum);
    writeln('Максимальный элемент: ', max);
    writeln('Среднее значение: ', sum / 10);
end.